docs/apache-airflow/concepts/deferring.rst

```diff
@@ -0,0 +1,172 @@
+* The new Trigger instance is registered inside Airflow, and picked up by one or more *triggerer* processes
+* The trigger is run until it fires, at which point its source task is re-scheduled
+* The task instance resumes
+
+Using deferrable operators as a DAG author is almost transparent; writing them, however, takes a bit more work.
+
+.. note::
+
+    Deferrable Operators & Triggers rely on more recent ``asyncio`` features, and as a result only work
+    on Python 3.7 or higher.
+
+
+Using Deferrable Operators
+--------------------------
+
+If all you wish to do is use pre-written Deferrable Operators (such as ``TimeSensorAsync``, which comes with Airflow), then there are only two steps you need:
+
+* Ensure your Airflow installation is running at least one *triggerer* process, as well as the normal *scheduler*
+* Use deferrable operators/sensors in your DAGs
+
+That's it; everything else will be automatically handled for you. If you're upgrading existing DAGs, we even provide some API-compatible sensor variants (e.g. ``TimeSensorAsync`` for ``TimeSensor``) that you can swap into your DAG with no other changes required.
+
+Note that you cannot yet use the deferral ability from inside custom PythonOperator/TaskFlow code; it is only available to pre-built Operators at the moment.
+
+
+Writing Deferrable Operators
+----------------------------
+
+Writing a deferrable operator takes a bit more work. There are some main points to consider:
+
+* Your Operator must defer itself based on a Trigger. If there is a Trigger in core Airflow you can use, great; otherwise, you will have to write one.
+* Your Operator will be deleted and removed from its worker while deferred, and no state will persist automatically. You can persist state by asking Airflow to resume you at a certain method or pass certain kwargs, but that's it.
+* You can defer multiple times, and you can defer before/after your Operator does significant work, or only defer if certain conditions are met (e.g. a system does not have an immediate answer). Deferral is entirely under your control.
+* Any Operator can defer; no special marking on its class is needed, and it's not limited to Sensors.
+
+
+Triggering Deferral
+~~~~~~~~~~~~~~~~~~~
+
+If you want to trigger deferral, at any place in your Operator you can call ``self.defer(trigger, method_name, kwargs, timeout)``, which will raise a special exception that Airflow will catch. The arguments are:
+
+* ``trigger``: An instance of a Trigger that you wish to defer on. It will be serialized into the database.
+* ``method_name``: The method name on your Operator you want Airflow to call when it resumes, other than ``execute``.
+* ``kwargs``: Additional keyword arguments to pass to the method when it is called. Optional, defaults to ``{}``.
+* ``timeout``: A timedelta that specifies a timeout after which this deferral will fail, and fail the task instance. Optional, defaults to ``None``, meaning no timeout.
```

Comment:
We need to document how this timeout and the existing execution_timeout interact.

(I _think_ that `execution_timeout` should be a total ceiling of runtime, i.e. time since task first started, right?)

Reply:
I think that is how I would interpret it too, and I suspect a bit of work is needed on that front.